dbos/system_database.go

```diff
@@ -671,6 +728,184 @@ func (s *sysDB) cancelWorkflow(ctx context.Context, workflowID string) error {
 	return nil
 }
 
+func (s *sysDB) resumeWorkflow(ctx context.Context, workflowID string) error {
+	tx, err := s.pool.Begin(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to begin transaction: %w", err)
+	}
+	defer tx.Rollback(ctx)
+
+	// Execute with snapshot isolation in case of concurrent calls on the same workflow
+	_, err = tx.Exec(ctx, "SET TRANSACTION ISOLATION LEVEL REPEATABLE READ")
+	if err != nil {
+		return fmt.Errorf("failed to set transaction isolation level: %w", err)
+	}
+
+	// Check the status of the workflow. If it is complete, do nothing.
+	listInput := listWorkflowsDBInput{
+		workflowIDs: []string{workflowID},
+		loadInput:   true,
+		loadOutput:  true,
+		tx:          tx,
+	}
+	wfs, err := s.listWorkflows(ctx, listInput)
+	if err != nil {
+		s.logger.Error("ResumeWorkflow: failed to list workflows", "error", err)
+		return err
+	}
+	if len(wfs) == 0 {
+		return newNonExistentWorkflowError(workflowID)
+	}
+
+	wf := wfs[0]
+	if wf.Status == WorkflowStatusSuccess || wf.Status == WorkflowStatusError {
+		return nil // Workflow is complete, do nothing
+	}
+
+	// Set the workflow's status to ENQUEUED and clear its recovery attempts and deadline
+	updateStatusQuery := `UPDATE dbos.workflow_status
+						  SET status = $1, queue_name = $2, recovery_attempts = $3, 
+						      workflow_deadline_epoch_ms = NULL, deduplication_id = NULL, 
+						      started_at_epoch_ms = NULL, updated_at = $4
+						  WHERE workflow_uuid = $5`
+
+	_, err = tx.Exec(ctx, updateStatusQuery,
+		WorkflowStatusEnqueued,
+		_DBOS_INTERNAL_QUEUE_NAME,
+		0,
+		time.Now().UnixMilli(),
+		workflowID)
```

Comment:
I am noticing that this resets the deadline (but keeps the timeout). In the current Go implementation, the timeout is not considered -- we only look at the deadline -- when running the workflow. I will fix this in a future PR.

Reply:
Resetting the deadline but keeping the timeout is the intended behavior of resume. The workflow should calculate a new deadline when it runs again (and if it doesn't, where are deadlines coming from?)

Reply:
right now the deadlines are only coming from:
- The context if first run/enqueue
- The deadline field in the DB if the workflow already exists

Recomputing the deadline is only needed when the workflow is cancelled/resumed. In all other paths (e.g., recovery), the original deadline is the one to enforce.

I think the best way to do this will be to just recompute the deadline from the timeout in the resume operation.